File: library-core/src/main/java/com/mikepenz/fastadapter/IInterceptor.java
package com.mikepenz.fastadapter;

import javax.annotation.Nullable;

/**
 * Created by mikepenz on 30.12.15.
 */
public interface IInterceptor<Element, Item> {

    IInterceptor<IItem, IItem> DEFAULT = new IInterceptor<IItem, IItem>() {
        @Override
        public IItem intercept(IItem item) {
            return item;
        }
    };

    @Nullable
    Item intercept(Element element);
}

File: library-core/src/main/java/com/mikepenz/fastadapter/IInterceptor.kt
package com.mikepenz.fastadapter

import android.support.v7.widget.RecyclerView

/**
 * Created by mikepenz on 30.12.15.
 */
interface IInterceptor<Element, Item> {

    fun intercept(element: Element): Item?

    companion object {
        val DEFAULT: IInterceptor<IItem<out RecyclerView.ViewHolder>, IItem<out RecyclerView.ViewHolder>> =
                object :
                        IInterceptor<IItem<out RecyclerView.ViewHolder>, IItem<out RecyclerView.ViewHolder>> {
                    override fun intercept(element: IItem<out RecyclerView.ViewHolder>): IItem<out RecyclerView.ViewHolder>? {
                        return element
                    }
                }
    }
}
